wikipedia/track.py

```diff
@@ -69,43 +80,44 @@ def __init__(self, track, params):
         self.ruleset_size = params.get("ruleset_size")
 
 
-class QueryIteratorParamSource(ParamSource):
+class RandomQueriesParamSource(ParamSource, metaclass=abc.ABCMeta):
     def __init__(self, track, params, **kwargs):
         super().__init__(track, params, **kwargs)
-        self._batch_size = self._params.get("batch_size", 100000)
-        self._random_seed = self._params.get("seed", None)
-        self._sample_queries = query_samples(self._batch_size, self._random_seed)
-        self._queries_iterator = None
+        self._queries_it = itertools.cycle(query_samples(k=self._params.get("batch_size", 100000), seed=self._params.get("seed")))
```

Comment:
Back to @gareth-ellis point I think this is correct, i.e. `itertools.cycle` will cycle back to the beginning of an array returned by `query_samples()` which is what current code is doing. It's convenient at the cost of doubling memory consumption because `itertools.cycle` stores a local copy of an array I think (see https://docs.python.org/3/library/itertools.html#itertools.cycle).

The `queries.csv` has 10k lines, and we're asking for x10 more as the default `batch_size/k` is 100k. Assuming pure ASCII (1 byte / character), and string length of 32 characters on average, that's an increase in the region of 128k * 32 ~ 4MB. I think that's OK.